plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/N4JSDeclaredNameValidator.xtend

```diff
@@ -322,6 +322,17 @@ class N4JSDeclaredNameValidator extends AbstractN4JSDeclarativeValidator {
 									 */
 									return;
 								} else {
+									if (baseEO instanceof Variable && dupeEO instanceof N4TypeDeclaration) {
+										if (N4JSLanguageUtils.isHollowElement(dupeEO as N4TypeDeclaration, jsVariantHelper)) {
+											return;
+										}
+									}
+									if (dupeEO instanceof Variable && baseEO instanceof N4TypeDeclaration) {
+										if (N4JSLanguageUtils.isHollowElement(baseEO as N4TypeDeclaration, jsVariantHelper)) {
+											return;
+										}
+									}
```

Comment:
This change allows having a hollow type and value of the same name. Currently this also applies to N4JS files. Wasn't the plan to only allow it in N4JSD files, for now?